packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue

```diff
@@ -280,106 +328,138 @@ const addConstraintForm = () => {
 function getStepList() {
-	const aList = [startTime.value];
-	const stepSize = floor((endTime.value - startTime.value) / numberOfSteps.value);
-	for (let i = 1; i < numberOfSteps.value; i++) {
+	const start = knobs.value.currentTimespan.start;
+	const end = knobs.value.currentTimespan.end;
+	const steps = knobs.value.numberOfSteps;
+
+	const aList = [start];
+	const stepSize = floor((end - start) / steps);
+	for (let i = 1; i < steps; i++) {
 		aList[i] = i * stepSize;
 	}
-	aList.push(endTime.value);
+	aList.push(end);
 	return aList;
 }
 
-const setModelOptions = async () => {
+const initialize = async () => {
 	const modelConfigurationId = props.node.inputs[0].value?.[0];
-	if (modelConfigurationId) {
-		modelConfiguration.value = await getModelConfigurationById(modelConfigurationId);
-		if (modelConfiguration.value) {
-			model.value = await getModel(modelConfiguration.value.modelId);
-			const modelColumnNameOptions: string[] =
-				modelConfiguration.value.configuration.model.states.map((state) => state.id);
-			// observables are not currently supported
-			// if (modelConfiguration.value.configuration.semantics?.ode?.observables) {
-			// 	modelConfiguration.value.configuration.semantics.ode.observables.forEach((o) => {
-			// 		modelColumnNameOptions.push(o.id);
-			// 	});
-			// }
-			modelNodeOptions.value = modelColumnNameOptions;
-
-			if (model.value && model.value.semantics?.ode.parameters) {
-				setRequestParameters(model.value.semantics?.ode.parameters);
-			} else {
-				toast.error('', 'Provided model has no parameters');
-			}
-		}
+	if (!modelConfigurationId) return;
+	modelConfiguration.value = await getModelConfigurationById(modelConfigurationId);
+	model.value = modelConfiguration.value.configuration as Model;
+};
+
+const setModelOptions = async () => {
+	if (!model.value) return;
+
+	const modelColumnNameOptions: string[] = model.value.model.states.map((state: any) => state.id);
+	// observables are not currently supported
+	// if (modelConfiguration.value.configuration.semantics?.ode?.observables) {
+	// 	modelConfiguration.value.configuration.semantics.ode.observables.forEach((o) => {
+	// 		modelColumnNameOptions.push(o.id);
+	// 	});
+	// }
+	modelNodeOptions.value = modelColumnNameOptions;
+
+	const state = props.node.state;
+	knobs.value.numberOfSteps = state.numSteps;
+	knobs.value.currentTimespan = _.cloneDeep(state.currentTimespan);
+	knobs.value.tolerance = state.tolerance;
+
+	if (model.value.semantics?.ode.parameters) {
+		setRequestParameters(model.value.semantics?.ode.parameters);
+
+		variablesOfInterest.value = requestParameters.value
+			.filter((d: any) => d.label === 'all')
+			.map((d: any) => d.name);
+	} else {
+		toast.error('', 'Provided model has no parameters');
 	}
 };
 
-const setRequestParameters = async (modelParameters) => {
+const setRequestParameters = (modelParameters: ModelParameter[]) => {
+	if (props.node.state.requestParameters) {
```

Comment:
Need to check if `requestParameters` is not empty.
node is initiated with state.requestParameters = [] which will satisfy this if

Reply:
Yeah, I just noticed this as well - nice catch.